File: admin-ui/lib/api/types/casa-de-paz-analytics.ts
export interface CasaDePazAnalyticsRange {
  from: string
  to: string
}

export interface CasaDePazAnalyticsTotals {
  houses: number
  adults: number
  kids: number
  guests: number
  /** adults + kids + guests — every person reached across all visits. */
  lives: number
  conversions: number
  conversion_rate: number
}

export interface CasaDePazAnalyticsGrowth {
  /** Percent change vs. the immediately preceding period of equal length.
   * `null` when there's no previous-period data to compare against. */
  houses: number | null
  lives: number | null
  guests: number | null
  conversions: number | null
}

export interface CasaDePazSeriesPoint {
  period: string
  houses: number
  adults: number
  kids: number
  guests: number
  conversions: number
}

export interface CasaDePazBySector {
  label: string
  sector_id: number | null
  houses: number
  adults: number
  kids: number
  guests: number
  conversions: number
}

export interface CasaDePazByDay {
  label: string
  houses: number
  adults: number
  guests: number
  conversions: number
}

export interface CasaDePazByTime {
  label: string
  houses: number
  adults: number
  guests: number
  conversions: number
}

export interface CasaDePazAnalyticsComparison {
  /** "YYYY-MM" of the most recent month with data used for growth comparisons. */
  period: string
}

export interface CasaDePazAnalyticsSummary {
  range: CasaDePazAnalyticsRange
  totals: CasaDePazAnalyticsTotals
  growth: CasaDePazAnalyticsGrowth
  /** `null` when there's no prior month with data to compare against. */
  comparison: CasaDePazAnalyticsComparison | null
  series: CasaDePazSeriesPoint[]
  by_sector: CasaDePazBySector[]
  by_day: CasaDePazByDay[]
  by_time: CasaDePazByTime[]
}

export interface CasaDePazAnalyticsSummaryQuery {
  /** "YYYY-MM-DD" — omit to default to 6 months back (start of month). */
  from?: string
  /** "YYYY-MM-DD" — omit to default to today. */
  to?: string
}

// Raw submission shape returned by the generic forms endpoint
// (GET /api/forms/casa-de-paz-reports), used for the drill-down table —
// mirrors backend/src/casa-de-paz-reports/entities/casa-de-paz-report.entity.ts
export interface CasaDePazReportSubmission {
  id: string
  date: string
  facilitator: string
  sector_id: number
  adults: number
  kids: number
  guests: number
  conversions: number
  meeting_day: string | null
  meeting_time: string | null
  created_at: string
  updated_at: string
}

export interface UpdateCasaDePazReportRequest {
  date?: string
  facilitator?: string
  sector_id?: number
  adults?: number
  kids?: number
  guests?: number
  conversions?: number
  meeting_day?: string
  meeting_time?: string
}

File: kmp-mobile/shared/src/commonMain/kotlin/br/church/paz/shared/domain/model/CasaDePazAnalytics.kt
package br.church.paz.shared.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class CasaDePazAnalyticsRange(
    val from: String,
    val to: String,
)

@Serializable
data class CasaDePazAnalyticsTotals(
    val houses: Int,
    val kids: Int,
    val guests: Int,
    // kids + guests — every person reached across all visits.
    val lives: Int,
    val conversions: Int,
    val conversionRate: Double,
)

@Serializable
data class CasaDePazAnalyticsGrowth(
    // Percent change vs. the most recent prior month with data. `null` when
    // there is no previous-period data to compare against — render no badge.
    val houses: Double?,
    val lives: Double?,
    val guests: Double?,
    val conversions: Double?,
)

@Serializable
data class CasaDePazSeriesPoint(
    val period: String,
    val houses: Int,
    val kids: Int,
    val guests: Int,
    val conversions: Int,
)

@Serializable
data class CasaDePazBySector(
    val label: String,
    val sectorId: Int?,
    val houses: Int,
    val kids: Int,
    val guests: Int,
    val conversions: Int,
)

@Serializable
data class CasaDePazByDay(
    val label: String,
    val houses: Int,
    val guests: Int,
    val conversions: Int,
)

@Serializable
data class CasaDePazAnalyticsComparison(
    // "YYYY-MM" of the most recent month with data used for growth comparisons.
    val period: String,
)

@Serializable
data class CasaDePazAnalyticsSummary(
    val range: CasaDePazAnalyticsRange,
    val totals: CasaDePazAnalyticsTotals,
    val growth: CasaDePazAnalyticsGrowth,
    // `null` when there's no prior month with data to compare against.
    val comparison: CasaDePazAnalyticsComparison?,
    val series: List<CasaDePazSeriesPoint>,
    val bySector: List<CasaDePazBySector>,
    val byDay: List<CasaDePazByDay>,
    // `by_time` is intentionally not modeled here — the "Horário" chart was
    // removed from admin-ui (commit 4494359) and must not be added on mobile.
)
